release/models/platform/openconfig-platform.yang

```diff
@@ -534,6 +541,89 @@ module openconfig-platform {
     }
   }
 
+  grouping platform-health-top {
+    description
+      "Grouping containing health-related parameters.";
+
+    container health {
+      description
+        "The health of the component. The paramaters within this
+        container indicate the status of the component beyond whether
+        it is operationally up or down. When a signal is received
+        that a component is in an unhealth state the gNOI.Healthz
+        service can be used to retrieve further diagnostic information
+        relating to the component.
+
+        The contents of this directory relate only to the specific
+        component that it is associated with. In the case that child
+        components become unhealthy and this causes a parent component
+        to be unhealthy, the new unhealthy status should be reported at
+        both components, such that an interested system can take the
+        relevant actions (e.g., retrieve the Healthz output, or
+        apply mitigation actions).";
+      reference
+        "https://https://github.com/openconfig/gnoi/tree/master/healthz";
+
+      container state {
+        config false;
+        description
+          "Operational state parameters relating to component health.";
+        uses platform-health-state;
+      }
+    }
+  }
+
+  grouping platform-health-state {
+    description
+      "Operational state parmaeters relating to a platform component's
+      health.";
+
+    leaf status {
+      type enumeration {
+        enum HEALTHY {
+          description
+            "The component is in a HEALTHY state, and is operating
+            within the expected parameters.";
+        }
+        enum DEGRADED {
+          description
+            "The component is in a degraded state. That is to say that
+            it has not completely failed, but failures have occurred that
+            either reduce its ability to perform, or may cause it to reach
+            an unhealthy state should further failures occur.
```

Comment:
a slight nit.  DEGRADED is still in an unhealthy state.  which is reinforced with the `last-unhealthy` treatment below.
potentially change to:

... or may cause it to reach a FAILED state should further failures or degradation occur.

Reply:
Removed `DEGRADED` per discussion in other comments.